dotrix_core/src/renderer/bindings.rs

```diff
@@ -76,25 +86,41 @@ impl<'a> BindGroup<'a> {
                     visibility: stage.into(),
                     ty: wgpu::BindingType::Texture {
                         multisampled: false,
-                        sample_type: wgpu::TextureSampleType::Float {
-                            filterable: texture.is_filterable(),
-                        },
+                        sample_type: texture.sample_type(),
                         view_dimension: wgpu::TextureViewDimension::D2,
                     },
                     count: None,
                 },
-                Binding::Texture3D(_, stage, texture) => wgpu::BindGroupLayoutEntry {
+                Binding::TextureCube(_, stage, texture) => wgpu::BindGroupLayoutEntry {
                     binding: index as u32,
                     visibility: stage.into(),
                     ty: wgpu::BindingType::Texture {
                         multisampled: false,
-                        sample_type: wgpu::TextureSampleType::Float {
-                            filterable: texture.is_filterable(),
-                        },
+                        sample_type: texture.sample_type(),
                         view_dimension: wgpu::TextureViewDimension::Cube,
                     },
                     count: None,
                 },
+                Binding::TextureArray(_, stage, texture) => wgpu::BindGroupLayoutEntry {
```

Comment:
[question]
Have you ever used texture arrays already?

Reply:
I'm working on using them at the moment. I am texturing the voxels which means a maximum of 256 materials each with 5 textures. Texture arrays just seem the best way to do this.

I am also considering changing how the we binds pbr textures rather than 5 bindings one for each texture, albedo, bump, etc an array might be best to just use a single texture array.

Reply:
I don't understand one thing with Texture Arrays and WGPU. For each array element we crate a `TextureView`. What `TextureView` should be used to Bind the whole Array then?

Reply:
Wait multiple TextureView? I thought there was just one for all of them. I thought it was like a 3D texture but without the interpolation along the z-axis. I'll finish my work voxel texturing and put it through it's passes there.

Reply:
You can create more TextureViews to access specific layer of the array and bind it as a single texture. I am still think about the API for it. No need to solve it in this PR.